src/server/game/AI/ScriptedAI/ScriptedEscortAI.cpp

```diff
@@ -298,11 +302,17 @@ void npc_escortAI::UpdateAI(uint32 diff)
         {
             if (DespawnAtFar && !IsPlayerOrGroupInRange())
             {
-                if (m_bCanInstantRespawn)
+                bool isEscort = false;
+                if (CreatureData const* cdata = me->GetCreatureData())
+                    isEscort = (sWorld->getBoolConfig(CONFIG_RESPAWN_DYNAMIC_ESCORTNPC) && (cdata->spawnGroupData->flags & SPAWNGROUP_FLAG_ESCORTQUESTNPC));
```

Comment:
Is there no helper for flags here? Like HasSpawnGroupFlags()?

Otherwise we should create that cuz flags & xyz is not nice.

Not needed to ve done in this PR but would be nice in the future

Reply:
Noted. I will look later.